include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,690 @@
+    size() const noexcept
+    {
+        return dn_;
+    }
+
+    /** Return true the string is empty
+    */
+    bool
+    empty() const noexcept
+    {
+        return n_ == 0;
+    }
+
+    /** Copy a decoded substring to another character string
+
+        This function copies a substring to the
+        character array pointed to by the
+        destination char string, where `rcount`
+        is the smaller of `count` and
+        `size() - pos`.
+
+        @par Exception Safety
+        Strong guarantee.
+        Exceptions thrown on invalid positions.
+
+        @throw std::out_of_range `pos > size()`
+
+        @param dest pointer to the destination character string
+        @param count requested substring length
+        @param pos position of the first character
+
+        @return Number of characters copied
+    */
+    BOOST_URL_DECL
+    size_type
+    copy(char* dest,
+         size_type count,
+         size_type pos = 0) const;
+
+    /** Return the result of comparing to another string
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.data(), rlen)`.
+        This means the comparison is performed with
+        percent-decoding applied to the current string.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(string_view other) const noexcept;
+
+    /** Return the result of comparing to another string
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.to_string().data(), rlen)`.
+        This means the comparison is performed with
+        percent-decoding applied to the current string.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(decoded_view other) const noexcept;
+
+    /** Returns a string with the decoded data
+    */
+    std::string
+    to_string() const
+    {
+        std::string r;
+        assign_to(r);
+        return r;
+    }
+
+    /** Assigns the decoded characters to a container
+
+        This function allows the caller to recycle
+        the input string.
+
+        @return A string representing the
+        entire contents of the decoded range.
+    */
+    template <class MutableString>
```

Comment:
You need to add a Concept to the docs, see Beast

Reply:
Done. However, I just noticed `vector<char>` won't have `append`. Maybe we can replace the constraint `s.append(first, last)` with another constraint `s.insert(s.end(), first, last)`.

Reply:
lets leave `vector<char>` out of it for now, and see what people say?